File: workspaces/ws_005_pomodorofocus/ios/PomodoroFocusTimer/Models/Session.swift
//
//  Session.swift
//  PomodoroFocusTimer
//
//  Plain value type mirroring the Core Data `SessionEntity`.
//  Data model: Session(id, date, type, durationSeconds, completedAt).
//

import Foundation

/// A completed (or in-progress, when used transiently) Pomodoro session.
struct Session: Identifiable, Hashable {
    let id: UUID
    /// Calendar day the session belongs to (start-of-day, used for grouping).
    let date: Date
    /// Phase type: focus / short / long.
    let type: PomodoroPhase
    /// Configured duration of the session in seconds.
    let durationSeconds: Int
    /// Exact timestamp the session finished.
    let completedAt: Date

    init(id: UUID = UUID(),
         date: Date,
         type: PomodoroPhase,
         durationSeconds: Int,
         completedAt: Date) {
        self.id = id
        self.date = date
        self.type = type
        self.durationSeconds = durationSeconds
        self.completedAt = completedAt
    }

    /// Whole minutes of the session (rounded down).
    var minutes: Int { durationSeconds / 60 }
}

File: workspaces/ws_005_pomodorofocus/android/app/src/main/java/com/appfactory/pomodorofocus/data/Session.kt
package com.appfactory.pomodorofocus.data

import androidx.room.Entity
import androidx.room.PrimaryKey
import java.util.UUID

/**
 * A completed Pomodoro session.
 *
 * Data model (per spec): Session(id, date, type, durationSeconds, completedAt).
 *
 * - [date] is the start-of-day epoch millis for the day the session belongs to,
 *   used to group sessions for the statistics queries.
 * - [completedAt] is the exact epoch-millis timestamp the session finished.
 * - [type] is one of "focus" / "short" / "long".
 */
@Entity(tableName = "sessions")
data class Session(
    @PrimaryKey
    val id: String = UUID.randomUUID().toString(),
    val date: Long,
    val type: String,
    val durationSeconds: Int,
    val completedAt: Long
) {
    /** Whole minutes of the session. */
    val minutes: Int get() = durationSeconds / 60
}
